File: Greedy/1798.Maximum_Number_of_Consecutive_Values_You_Can_Make/C++/1798.Maximum_Number_of_Consecutive_Values_You_Can_Make.cpp
class Solution {
public:
    int getMaximumConsecutive(vector<int> & coins) 
    {
        //=====================================//
        // Input type:                         //
        //   - vector<int> ref. (coins vector) //
        // Return type:                        //
        //   - int (result maximun length)     //
        //=====================================//

        /*Initialize*/
        ///// Length of coins vector /////
        int len_coins = coins.size();

        ///// Record summary counter /////
        int record_cnter = 1;


        /*Greedy-based loop traversal with sorted-operations*/
        sort(coins.begin(), coins.end()); //Sorted-operations

        for (int coins_idx = 0; (coins_idx < len_coins); (coins_idx++)) //Whole
        {
            ///// Check if the current indexed-value matched conditions or not /////
            if (coins[coins_idx] <= record_cnter) { record_cnter += coins[coins_idx]; } //Keep updating, accumulating
            else { return record_cnter; }
        }

        return record_cnter;
    }
};

File: Greedy/1798.Maximum_Number_of_Consecutive_Values_You_Can_Make/Python/1798.Maximum_Number_of_Consecutive_Values_You_Can_Make.py
class Solution(object):
    def getMaximumConsecutive(self, coins):
        """
        :type coins: List[int]
        :rtype: int
        """
        #===============================#
        # Greedy-based traversal method #
        #===============================#

        ############
        #Initialize
        ##### Length of coins array #####
        len_coins = len(coins)

        ##### Record summary counter #####
        record_cnter = 1


        ####################################################
        #Greedy-based loop traversal with sorted-operations
        coins.sort() #Sorted-operations

        for coins_idx in range(len_coins):

            ##### Check if the current indexed-value matched conditions or not #####
            if (coins[coins_idx] <= record_cnter):
                record_cnter += coins[coins_idx] #Keep updating/accumulating
            else:
                return record_cnter

        return record_cnter
